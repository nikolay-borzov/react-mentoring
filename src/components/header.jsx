import React from 'react'
import PropTypes from 'prop-types'

export class Header extends React.Component {
  static propTypes = {
    children: PropTypes.any.isRequired
  }

  render() {
    return <header className="alt-background">{this.props.children}</header>
  }
}
