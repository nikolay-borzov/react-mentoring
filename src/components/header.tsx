import * as React from 'react'

interface HeaderProps {
  children: React.ReactNode
}

export const Header: React.FunctionComponent<HeaderProps> = props => {
  return <header className="alt-background">{props.children}</header>
}
