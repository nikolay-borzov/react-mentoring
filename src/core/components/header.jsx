import React from 'react'
import PropTypes from 'prop-types'

export class Header extends React.PureComponent {
  static propTypes = {
    children: PropTypes.element.isRequired
  }

  render() {
    return <header className="alt-background">{this.props.children}</header>
  }
}
