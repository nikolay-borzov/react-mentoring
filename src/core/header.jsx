import React from 'react'
import PropTypes from 'prop-types'

export class Header extends React.PureComponent {
  render() {
    return <header className="layout-block">{this.props.children}</header>
  }
}

Header.propTypes = {
  children: PropTypes.element.isRequired
}
