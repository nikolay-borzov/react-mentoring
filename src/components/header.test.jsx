import React from 'react'
import renderer from 'react-test-renderer'

import { Header } from './header'

describe('Header component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Header>
          Header content <p>More content</p>
        </Header>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
