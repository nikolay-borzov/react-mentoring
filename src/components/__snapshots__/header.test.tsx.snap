// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<header
  className="alt-background"
>
  Header content 
  <p>
    More content
  </p>
</header>
`;
